src/script/components/InputBar/InputBar.tsx

```diff
@@ -395,15 +396,15 @@ const InputBar = ({
   }, [editMessageEntity]); // eslint-disable-line react-hooks/exhaustive-deps
 
   useEffect(() => {
-    if (!hasUserTyped.current) {
+    if (!hasUserTyped.current || !typingIndicatorMode) {
```

Comment:
Since it's a mode, it's risky to just to a falsy/truthy check. 
Let's be explicit 

```suggestion
    if (!hasUserTyped.current || !typingIndicatorMode === CONVERSATION_TYPING_MODE.OFF) {
```